cmd/vela-docker/registry.go

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package main
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/go-vela/types/constants"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	credentials = `%s:%s`
+
+	registryFile = `{
+  "auths": {
+    "%s": {
+      "auth": "%s"
+    }
+  }
+}`
+
+	// nolint
+	loginAction = "login"
+)
+
+// Registry represents the input parameters for the plugin.
+type Registry struct {
+	// enable building the image without publishing
+	DryRun bool
+	// full url to Docker Registry
+	Name string
+	// password for communication with the Docker Registry
+	Password string
+	// user name for communication with the Docker Registry
+	Username string
+}
+
+var (
+	// appFs represents a instance of the filesystem.
+	appFS = afero.NewOsFs()
+
+	// registryFlags represents for registry settings on the cli.
+	// nolint
+	registryFlags = []cli.Flag{
+		&cli.BoolFlag{
+			EnvVars:  []string{"PARAMETER_REGISTRY", "REGISTRY_DRY_RUN"},
```

Comment:
probably shouldn't be `PARAMETER_REGISTRY`?

Reply:
yeah that was a typo, fixing.